sequencer/src/block2/queryable.rs

```diff
@@ -46,15 +49,17 @@ impl QueryablePayload for Payload {
         // sum over these tx table lens
         let mut result = 0;
         for &offset in ns_end_offsets.iter().take(ns_end_offsets.len() - 1) {
-            let tx_table_len = get_table_len(&self.payload, offset).try_into().unwrap_or(0);
+            let tx_table = TxTable::<u32>::from_bytes(&self.raw_payload);
+            let tx_table_len = tx_table.get_table_len(offset).try_into().unwrap_or(0);
             // TODO handle large tx_table_len! (https://github.com/EspressoSystems/espresso-sequencer/issues/785)
             result += tx_table_len;
         }
         result
     }
 
     fn iter<'a>(&'a self, meta: &'a Self::Metadata) -> Self::Iter<'a> {
-        TxIterator::new(meta, self)
+        let ns_table = NameSpaceTable::from_bytes(meta);
```

Comment:
I see, as discussed above I think the best solution would be to store a reference to the payload bytes instead of cloning. Added a TODO in 0524f2f. Happy to keep discussing further though.